### DISCUSSION TITLE
Shortcut to add/insert semicolon to end of line

### DISCUSSION THREAD

**Discussion by Richard3134** [geany/geany#3596] [Q&A] [answered]
Hello,

I am looking online and in Geany to find a way to setup this shortcut: ad/insert semicolon to EOL, I couldn't find any clue... 

In advance, sorry if I missed something somewhere... Is anyone could help me with this?

Thanks a lot,
Richard

**Comment by elextr** [comment, marked as answer]
There is no built-in "add semicolon at EOL" action, so you can't just set a keybinding.  

You might be able to define a macro with suitable keystrokes using the geanymacro plugin and assign that a keybinding.

**Reply by Richard3134** [reply]
Many thanks elextr. The auto-close plugin adds semicolon while typing a known function, therefore it should be enough in most circumstances. I will implement your suggestion anyway because I am wondering if this plugin is not interfering with Firefox somehow. I have some keyboards issue in Firefox that I solve closing Geany....strange. Thanks a lot for your help. Regards. Richard
